src/main/java/com/iridium/iridiumskyblock/managers/IslandManager.java

```diff
@@ -690,19 +690,22 @@ public ItemStack getIslandCrystal(int amount) {
         ItemStack itemStack = ItemStackUtils.makeItem(IridiumSkyblock.getInstance().getConfiguration().islandCrystal, Collections.singletonList(
                 new Placeholder("amount", String.valueOf(amount))
         ));
-        NBTItem nbtItem = new NBTItem(itemStack);
-        NBTCompound nbtCompound = nbtItem.getOrCreateCompound("iridiumskyblock");
-        nbtCompound.setInteger("islandCrystals", amount);
-        return nbtItem.getItem();
+
+        NBT.modify(itemStack, readWriteItemNBT -> {
+            readWriteItemNBT.getOrCreateCompound("iridiumskyblock").setInteger("islandCrystals", amount);
+        });
+
+        return itemStack;
     }
 
     public int getIslandCrystals(ItemStack itemStack) {
         if (itemStack == null || itemStack.getType() == Material.AIR) return 0;
-        NBTCompound nbtCompound = new NBTItem(itemStack).getOrCreateCompound("iridiumskyblock");
-        if (nbtCompound.hasKey("islandCrystals")) {
-            return nbtCompound.getInteger("islandCrystals");
-        }
-        return 0;
+
+        return NBT.get(itemStack, readableItemNBT -> {
+            if(readableItemNBT.getCompound("iridiumskyblock").hasTag("islandCrystals"))
+                return readableItemNBT.getCompound("iridiumskyblock").getInteger("islandCrystals");
+            return 0;
```

Comment:
I think in general you should add brackets onto if statements unless its on the same line


e.g. the only place where its ok to not have brackets is something like

```
if (thing) return
```

if you did

``` if(thing)
return;
```

This shouldnt be allowed (its very easy for bugs to appear since it gets confusing whats in and whats not inside the if statement otherwise)

Reply:
roger doger